### PR TITLE
tests: make yarn unit run all available unit tests

### PR DESCRIPTION
**Pull request by brendankenny**
we keep growing `yarn unit` by adding every new `yarn unit-*`, but `jest.config.js` has to have references to these files anyways, so `yarn jest` works just fine if you just want to run all the unit tests :)

This also fixes some overlaps and cracks between the `yarn unit-*`s. The current `yarn unit` doesn't run
- ~`clients/test/extension/popup-test.js`~
- `clients/test/extension/settings-controller-test.js`
- `clients/test/lightrider-entry-test.js`
- ~`docs/recipes/integration-test/example-lh-auth.test.js`~
- `third-party/chromium-synchronization/inspector-issueAdded-types-test.js`
- `third-party/chromium-synchronization/installability-errors-test.js`

the last two might be on purpose but seems like they should be in the testing path if the goal is for changes in Canary to alert us to needed updates? It also made us miss that the path to `LH_ROOT` was wrong in `inspector-issueAdded-types-test.js` and it doesn't run at all.

We're also running the `flow-report` tests twice because they're picked up by both `yarn unit-report` and `yarn unit-flow`.

This is most important in CI, where unit.yml runs them via [`yarn unit:ci`](https://github.com/GoogleChrome/lighthouse/blob/35d5b214386ea31cdf814095e477e747c118c0bf/.github/workflows/unit.yml#L55) and should be running all the tests.

### CODE REVIEW

**Comment by brendankenny** [comment]
> * `clients/test/extension/popup-test.js`
> * `clients/test/extension/settings-controller-test.js`
> * `clients/test/lightrider-entry-test.js`
> * `docs/recipes/integration-test/example-lh-auth.test.js`

well, these _are_ covered by `yarn test-clients` and `yarn test-docs` in `ci.yml`. I'm pretty fine running them with `yarn unit` if they're going to be listed in the `jest.config.js`, though.

Context here is that https://github.com/GoogleChrome/lighthouse/pull/13146 is adding yet another directory with tests in it, and I don't really have any interest in adding a `yarn unit-shared` with bonus `&& yarn unit-shared` and `&& npm run unit-shared -- --ci` when `yarn unit` could just run them all and `yarn jest shared` works just fine for narrowing to just those tests already :)

File: .github/workflows/unit.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ jobs:
         pip install protobuf==3.7.1
 
     - run: yarn install --frozen-lockfile --network-timeout 1000000
-    - run: yarn build-report
```
Comment (adamraine) [inline .github/workflows/unit.yml:47]: What were we missing out on before this change?

Comment (brendankenny) [reply .github/workflows/unit.yml:47]: > What were we missing out on before this change?

tests under `clients/test/extension/` needed `yarn build-extension` and `lightrider-entry-test.js` needed `yarn build-lr` and with `yarn build-report` already needed, `build-all` just seemed easier :)

**Comment by brendankenny** [comment]
rather than make `docs/recipes/integration-test/example-lh-auth.test.js` work in the unit tests, would rather just remove it from the unit tests (#13150)

**Comment by brendankenny** [comment]
updated with docs test changes from #13150 so `yarn test-docs` doesn't run under `yarn unit`.

Also moved the `*-test-pptr.js` tests previously identified as flaky ([they have automatic retry in CI](https://github.com/GoogleChrome/lighthouse/blob/e6921d9b1980ad8b50a3ad1705ff1e2102487140/.github/workflows/ci.yml#L36-L43)) out of the default unit tests. They now only run under their respective `yarn test-clients`/`yarn test-viewer`/`yarn test-treemap` and not under `yarn unit`, which is exactly the current behavior.

`fraggle-rock/**/*-test-pptr.js` has been running in `unit.yml` without flakes for a while now, so seems ok to leave under unit tests for now.

**Review by adamraine — APPROVED** [review]

**Comment by Mbellsudteen** [comment]
ในวันที่ ศ. 1 ต.ค. 2021 04:19 น. devtools-bot ***@***.***>
เขียนว่า:

**Pull request merged**